src/functional/geometry/convex_hull.zig

```diff
@@ -0,0 +1,243 @@
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+    point.time = 9;
+    point.value = 3.5;
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+    point.time = 10;
+    point.value = 2.5;
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+    point.time = 11;
+    point.value = 2.5;
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+    point.time = 12;
+    point.value = 3.5;
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+    point.time = 13;
+    point.value = 2.5;
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+    point.time = 14;
+    point.value = 2.5;
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+    point.time = 15;
+    point.value = 2.5;
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+    point.time = 16;
+    point.value = 3;
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+    point.time = 17;
+    point.value = 3;
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+    point.time = 18;
+    point.value = 3;
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+    point.time = 19;
+    point.value = 3;
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+    point.time = 20;
+    point.value = 2.8;
+    try addPointToConvexHull(&upper_hull, &lower_hull, point);
+
+    try testing.expectEqual(5, upper_hull.len);
+    try testing.expectEqual(4, lower_hull.len);
+
+    // Expected Upper Hull.
+    try testing.expectEqual(0, upper_hull.points[0].time);
+    try testing.expectEqual(3, upper_hull.points[1].time);
+    try testing.expectEqual(8, upper_hull.points[2].time);
+    try testing.expectEqual(19, upper_hull.points[3].time);
+    try testing.expectEqual(20, upper_hull.points[4].time);
+    // Expected Lower Hull.
+    try testing.expectEqual(0, lower_hull.points[0].time);
+    try testing.expectEqual(1, lower_hull.points[1].time);
+    try testing.expectEqual(15, lower_hull.points[2].time);
+    try testing.expectEqual(20, lower_hull.points[3].time);
+}
+
+test "incremental convex hull random elements" {
+    const num_points: usize = 1000;
+    const allocator = testing.allocator;
+    var rnd = std.rand.DefaultPrng.init(0);
+    var upper_hull = try PointSet.init(&allocator, num_points);
+    defer upper_hull.deinit();
+    var lower_hull = try PointSet.init(&allocator, num_points);
+    defer lower_hull.deinit();
+
+    var point: Point = Point{ .time = 0, .value = rnd.random().float(f64) };
+    for (1..num_points) |i| {
+        try addPointToConvexHull(&upper_hull, &lower_hull, point);
+        point.time = i;
+        point.value = rnd.random().float(f64);
+    }
+
+    // All points in the Upper Hull should turn to the right.
+    for (1..upper_hull.len - 1) |i| {
+        const turn = getTurn(upper_hull.points[i - 1], upper_hull.points[i], upper_hull.points[i + 1]);
+        try testing.expectEqual(turn, Turn.right);
+    }
```

Comment:
Maybe add an empty line after this `}` to make it clear two different things are evaluated.

Reply:
Fixed.